GetMessages/handler.ts

```diff
@@ -46,18 +61,18 @@ export function GetMessagesHandler(
   messageModel: MessageModel
 ): IGetMessagesHandler {
   return async fiscalCode => {
-    const retrievedMessagesIterator = messageModel.findMessages(fiscalCode);
-    const validMessagesIterator = filterResultIterator(
-      retrievedMessagesIterator,
-      // isPending is true when the message has been received from the sender
-      // but it's still being processed
-      message => message.isPending !== true
-    );
-    const publicExtendedMessagesIterator = mapResultIterator(
-      validMessagesIterator,
-      retrievedMessageToPublic
-    );
-    return ResponseJsonIterator(publicExtendedMessagesIterator);
+    return messageModel
+      .findMessages(fiscalCode)
+      .map(flattenAsyncIterator)
+      .map(_ => filterAsyncIterator(_, isRight))
+      .map(_ => mapAsyncIterator(_, e => e.value))
+      .map(_ => filterAsyncIterator(_, RetrievedPendingMessage.is))
```

Comment:
isn't this

```suggestion
      .map(_ => filterAsyncIterator(_, not(RetrievedPendingMessage.is)))
```
?

Reply:
may we add a test for this case?

Reply:
see https://github.com/pagopa/io-functions-app/pull/99/commits/8966ba5a61283fe96138d1f868df51ea8b44a2ce